envelopes/bin/envelopes

```diff
@@ -6,27 +6,32 @@ module Envelopes
 
     desc "log_in NAME", "log in"
     def log_in(name)
-      Store.log_in(name)
+      CurrentUser.log_in(name)
+    end
+
+    desc "current_user", "returns the person logged in"
+    def current_user
+      puts CurrentUser.current_user
     end
     
     desc "create_envelope LIST", "create empty envelope"
     def create_envelope(*categories)
-      Budget.create_new_categories(categories)
+      Budget.create_new_categories(categories, current_user)
```

Comment:
The implications of adding a current user are kind of interesting.  It seems it requires you to pass it in everywhere.  I counted 17 methods that now need to have it passed in!

🤔

Reply:
yes!  I had this concern when I thought about creating the login-object... but what are other options?

Reply:
Ok right!
I think we can do one of two (interesting) things
we can have a global variable, `$current_user`
or we can pass in a default argument to all those methods.

```
def some_method(real_argument, current_user = CurrentUser.current_user)
  # stuff here
end
```

Reply:
the `$global_variable` is usually not good practice in web work but in your CLI it would be fine because the whole program is spun up for each request.

Reply:
> Ok right!
> I think we can do one of two (interesting) things
> we can have a global variable, `$current_user`
> or we can pass in a default argument to all those methods.
> 
> ```
> def some_method(real_argument, current_user = CurrentUser.current_user)
>   # stuff here
> end
> ```
Would I write this function for ever function in the interface?
This feels similar to passing in the current_user to all the arguments..?

Reply:
> the `$global_variable` is usually not good practice in web work but in your CLI it would be fine because the whole program is spun up for each request.

The global variable would need to be checked/changed for every request... right?

Reply:
Yea, that's a good point.  🤔 
Yea, it seems like a cheat to get around the ugliness of passing it in, but it doesn't really get the abstraction of  a current user right.

So the global _might_ work better for this case.  
You're right, and if we're thinking web where requests are a thing this might be a problem.
In our current case each time you send a command in the CLI it will start the whole program again.  We can just set the global when the program starts each time no matter what command is run, then the whole program will have it.  After it is finished executing it just stop, so no problems I think.